hub/gads-ui/src/components/DeviceControl/StreamCanvas/StreamCanvas.js

```diff
@@ -45,6 +45,54 @@ export default function StreamCanvas({ deviceData, shouldShowStream }) {
         streamUrl = `/device/${deviceData.udid}/android-stream-mjpeg`
     }
 
+    useEffect(() => {
+        function handleKeyDown(event) {
+            // Check if there is an active element to avoid capturing keyboard events
+            // When we want to be typing somewhere else, not send to the device
+            const activeElement = document.activeElement
+            const isInputFocused = activeElement && (
+                activeElement.tagName === 'INPUT' ||
+                activeElement.tagName === 'TEXTAREA' ||
+                activeElement.isContentEditable
+            )
+
+            // Don't process keyboard events when user is typing somewhere
+            if (isInputFocused) {
+                return
+            }
+
+            // Handle normal chars, special chars, Enter and Backspace, ignore stuff like
+            //  Shift, Ctrl, Alt, F1-F12, etc.
+            const key = event.key
+            if (key.length === 1) {
+                sendKeyPress(key)
+            } else if (key === 'Enter') {
+                sendKeyPress('\n')
+            } else if (key === 'Backspace') {
+                sendKeyPress('\b')
+            } else {
+                return
+            }
```

Comment:
Explicitly document or account for edge cases where other special keys (e.g., Escape, Tab) might require processing. Ignoring them entirely could limit functionality.

Reply:
Tab and Escape are not native keys on phone keyboards, I am trying to emulate typing to the extent that you might need from a phone keyboard but not a hardware one. Also handling Escape might be kind of feasible for Android but it is not for iOS. The current WebDriverAgent endpoint supports literal characters/strings, not actual keys. For Enter/Backspace we handle `\n` and `\b` respectively, there is no such alternative for Escape. I would rather keep this as is and extend if needed or at all possible in the future :)